tests/test_server.py

```diff
@@ -316,6 +318,25 @@ def version(self):
 
     print(MockServer(remote_server).version)
     with pytest.raises(
-        ValueError, match="Error connecting to DPF LegacyGrpcServer with version 1.0"
+            ValueError, match="Could not connect to DPF legacy gRPC server with version 1.0."
     ):
         dpf.core.server_types.check_ansys_grpc_dpf_version(MockServer(remote_server), timeout=2.0)
+
+
+@pytest.mark.skipif(
+    running_docker,
+    reason="can only work with local DPF server install",
+)
+def test_available_servers():
+    out = server.available_servers()
+    assert len(out) > 0
+    for version in out:
+        vout = str(version).replace(".", "_")
+        if packaging.version.parse(version) > packaging.version.parse("2022.1"):
+            meth = "start_" + vout + "_server"
+            assert (hasattr(server, meth))
+            start = getattr(server, meth)
+            srv = start()
+            assert srv.local_server
+            srv = out[version]()
+            assert srv.local_server
```

Comment:
Hey @rlagha,
thanks for the helper, this is great. I guess we should also test that the server version ran is the correct one.
```suggestion
            assert srv.local_server
            assert srv.version == version
            srv = out[version]()
            assert srv.local_server
            assert srv.version == version
```